File: pi-backend/Templateexamen23-24/src/main/java/tn/examen/templateexamen2324/entity/RequestSupply.java
package tn.examen.templateexamen2324.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class RequestSupply {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  int idRequestSupply;
    public int quantity;
    public String category;
    public  String description;
    public LocalDate date;
    public int validity;
    @Enumerated(EnumType.STRING)
    private RequestSupplyStatus status = RequestSupplyStatus.Running;
    @OneToMany(cascade = CascadeType.ALL,mappedBy = "requestSupply")
    @JsonIgnore
    private Set<Devis> devis = new HashSet<>();
    @OneToOne(cascade = CascadeType.ALL)
    @JsonIgnore
    private Invoice invoice;
    @ManyToOne(cascade = CascadeType.ALL)
    @JsonIgnore
    private Individu individu;
    @ManyToOne(fetch = FetchType.EAGER,cascade = CascadeType.ALL)
    //@JsonIgnore
    private Forum forum;
}

File: frontend/pi-backOffice/src/app/model/requestSupply.ts
import { Devis } from "./devis";
import { Forum } from "./forum";
import { Individu } from "./individus";
import { Invoice } from "./invoice";

export class RequestSupply {
    idRequestSupply!: number;
    quantity!: number;
    category!: string;
    description!: string;
    date!: Date;
    validity!: number;
    status!: String;
    devis!: Devis[];
    invoice!: Invoice;
    individu!: Individu;
    forum!:Forum;
    RequestSuplyStatus!: string;
}